Adding StackExchange/blackbox v1.20150730

booi opened the pull request.

dunn commented:
`brew audit` flagged a few issues:
```
== /usr/local/Library/Formula/blackbox.rb ==
C:  8: 21: Trailing whitespace detected.
C: 34:  1: Incorrect indentation detected (column 0 instead of 2).
C: 36:  1: 1 trailing blank lines detected.

1 file inspected, 3 offenses detected
Error: 6 problems in 1 formula

==> audit problems
blackbox:
 * A `test do` test block should be added
 * Stable: version 1.20150730 is redundant with version scanned from URL
 * Formula should have a desc (Description).
 * 8: Trailing whitespace was found
 * Non-executables were installed to "/usr/local/Cellar/blackbox/1.20150730_1/bin"
The offending files are:
  /usr/local/Cellar/blackbox/1.20150730_1/bin/_stack_lib.sh
 * blackbox seems to be listed in tap_migrations.rb!
Please remove blackbox from present tap & tap_migrations.rb
before submitting it to Homebrew/homebrew.
```

DomT4 commented on Library/Formula/blackbox.rb:
```diff
@@ -0,0 +1,36 @@
+  url "https://github.com/StackExchange/blackbox/archive/v1.20150730.tar.gz"
+  version "1.20150730"
+  sha256 "a28237e8839d000273d0fe0ba8f9701cab66a3b942787c5c8e8d329912564ae4"
+  revision 1
```
The default revision is `0`, and unwritten. You can cut this line.

booi replied:
Ok I will remove.

DomT4 commented on Library/Formula/blackbox.rb:
```diff
@@ -0,0 +1,36 @@
+    bin.install "bin/blackbox_whatsnew"
+  end
+
+# Unfortunately we can't run the tests "as is" if we have a ~/.gnupg directory.
```
You won't have a `~/.gnupg` directory, ever, in this test. For the `test do` block we temporarily divert `$HOME` to point at the `testpath`.

booi replied:
Ok, let me try to make some rudimentary functionality tests.

DomT4 commented on Library/Formula/blackbox.rb:
```diff
@@ -0,0 +1,36 @@
+  depends_on "gnupg" 
+
+  def install
+    bin.install "bin/_blackbox_common.sh"
```
Is there anything in `bin/` that we *don't* want to install, out of curiosity?

booi replied:
Probably not. We want everything but I didn't want to potentially overwrite anything in case of an upstream change. 

> On Sep 26, 2015, at 8:58 PM, Dominyk Tiller <notifications@github.com> wrote:
> 
> In Library/Formula/blackbox.rb:
> 
> > @@ -0,0 +1,36 @@
> > +class Blackbox < Formula
> > +  homepage "http://the-cloud-book.com/"
> > +  url "https://github.com/StackExchange/blackbox/archive/v1.20150730.tar.gz"
> > +  version "1.20150730"
> > +  sha256 "a28237e8839d000273d0fe0ba8f9701cab66a3b942787c5c8e8d329912564ae4"
> > +  revision 1
> > +
> > +  depends_on "gnupg" 
> > +
> > +  def install
> > +    bin.install "bin/_blackbox_common.sh"
> Is there anything in bin/ that we don't want to install, out of curiosity?
> 
> —
> Reply to this email directly or view it on GitHub.
>

DomT4 replied:
If we want everything for now, let's just do `bin.install Dir["bin/*"]`

booi replied:
Ok sounds good, I'll put a note for future upstream changes.

xu-cheng commented on Library/Formula/blackbox.rb:
```diff
@@ -0,0 +1,36 @@
+  sha256 "a28237e8839d000273d0fe0ba8f9701cab66a3b942787c5c8e8d329912564ae4"
+  revision 1
+
+  depends_on "gnupg" 
```
Should be `depends_on :gpg` if it works with gpg2 too.

booi replied:
Yes it works for gpg2. I'll put that in.

DomT4 commented:
Any progress here?

booi commented:
Hey guys, sorry I went on vacation and forgot about this. I've cleaned up the formula as requested and moved the commit into a new pull request https://github.com/Homebrew/homebrew/pull/45253.
